server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -54,22 +56,31 @@
     public static final String NAME = "auto_date_histogram";
 
     private static final ParseField NUM_BUCKETS_FIELD = new ParseField("buckets");
+    private static final ParseField MINIMUM_INTERVAL_FIELD = new ParseField("minimum_interval");
 
     private static final ObjectParser<AutoDateHistogramAggregationBuilder, Void> PARSER;
     static {
         PARSER = new ObjectParser<>(AutoDateHistogramAggregationBuilder.NAME);
         ValuesSourceParserHelper.declareNumericFields(PARSER, true, true, true);
-
         PARSER.declareInt(AutoDateHistogramAggregationBuilder::setNumBuckets, NUM_BUCKETS_FIELD);
+        PARSER.declareString(AutoDateHistogramAggregationBuilder::setMinimumIntervalExpression, MINIMUM_INTERVAL_FIELD);
     }
 
+    private static List<String> allowedIntervals = Arrays.asList("second", "minute", "hour", "day", "month", "year");
```

Comment:
thought about using an enum here, didn't seem to buy too much.

Reply:
Could possibly re-use `Rounding.DateTimeUnit` enum, although that also includes `quarter`.  And would have to add string representations to the enum since it doesn't right now.  Might not be worth the hassle.  Another candidate is `DateHistogramAggregationBuilder.DATE_FIELD_UNITS`, although that also includes `1w`, etc.

Although it does make me wonder, was `quarter` skipped on purpose and/or some reason to not have it?

EDIT: Might be moot given Colin's comment below